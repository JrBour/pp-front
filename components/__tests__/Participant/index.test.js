import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Participant from '@/components/Participant'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Participant', () => {
  let wrapper
  let store
  let state

  beforeEach(() => {
    state = {
      participants: [
        {
          id: '1'
        }
      ]
    }
    store = new Vuex.Store({ state })
    wrapper = shallowMount(Participant, {
      localVue,
      store,
      propsData: {
        user: {
          givenName: 'John',
          lastName: 'Doe',
          id: '1'
        },
        displayAddButton: true
      },
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
