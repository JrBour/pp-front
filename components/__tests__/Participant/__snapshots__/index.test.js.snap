// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant snapshot renders properly 1`] = `
<div class="participant">
  <div class="participant__wrapper">
    <div style="background-image: url(); background-repeat: no-repeat; background-size: cover; background-position: center;"></div>
    <p>John Doe</p>
  </div> <button>
    +
  </button>
</div>
`;
